weave

```diff
@@ -1438,11 +1508,25 @@ case "$COMMAND" in
     run)
         dns_args "$@"
         shift $(dns_arg_count "$@")
+        dns_extra_hosts_args "$@"
         collect_cidr_args "$@"
         shift $CIDR_ARG_COUNT
+        REWRITE_HOSTS=1
+        while [ $# -gt 0 ]; do
+            case "$1" in
+                --no-rewrite-hosts)
+                    REWRITE_HOSTS=
+                    ;;
+                *)
+                    break
+                    ;;
+            esac
+            shift
+        done
```

Comment:
This needs to happen before the `collect_cidr_args` call, otherwise that will not see CIDRS after a `--no-rewrite-hosts`. Also, there is no need for a loop.

I'd also defer the `dns_extra_hosts_args` call until we need it in the `-n "$REWRITE_HOSTS"` path (only) below.